eclass/xdg.eclass

```diff
@@ -51,24 +52,35 @@ xdg_pkg_preinst() {
 		XDG_ECLASS_DESKTOPFILES+=( ${f} )
 	done < <(cd "${D}" && find 'usr/share/applications' -type f -print0 2>/dev/null)
 
+	XDG_ECLASS_ICONFILES=()
+	while IFS= read -r '' f; do
+		XDG_ECLASS_ICONFILES+=( ${f} )
+	done < <(cd "${D}" && find 'usr/share/icons' -type f -print0 2>/dev/null)
+
 	XDG_ECLASS_MIMEINFOFILES=()
 	while IFS= read -r -d '' f; do
 		XDG_ECLASS_MIMEINFOFILES+=( ${f} )
 	done < <(cd "${D}" && find 'usr/share/mime' -type f -print0 2>/dev/null)
 
-	export XDG_ECLASS_DESKTOPFILES XDG_ECLASS_MIMEINFOFILES
+	export XDG_ECLASS_DESKTOPFILES XDG_ECLASS_ICONFILES XDG_ECLASS_MIMEINFOFILES
```

Comment:
There is no reason to export these variables to the environment. That doesn't work properly for arrays anyway. Please remove this line.

Reply:
This used to be exported on purpose to get it from environment in postrm. I don't know what the story now is.

Reply:
You don't need to export the variables to access them from pkg_postrm. Portage will take care of saving/restoring default scope (non-local) variables as required by PMS.

Reply:
Regarding arrays, I see them exported into environment just fine. Haven't tested in practice specifically, but environment.bz2 has it with `declare -ax`.
Old code used to put the icon base paths (like `usr/share/icons/hicolor`) into the variable (space separate I believe), and then run icon cache update for only those icon themes. Since then, it had been simplified to just run it for all present icon themes instead, and in that case we could do it with just a boolean now. Though storing all icon themes where icons are installed would be more future proof. We certainly don't need an array of all icons, only the base icontheme paths like the old code seems to have behaved. I believe this new code saves a list of all icon files now, and we will never need that.

Reply:
It was newline separated, which old code before simplifications just looped with default $IFS (I think), so not so good for space containing EROOT, but could have been fixed with a $IFS without spaces (or arrays).

Reply:
I can only re-iterate that exporting them as environment variables in wrong and unnecessary. Portage will save/restore the variables either way.

I know what I'm talking about here.

Reply:
Either store a boolean-kind value or an array of icontheme base paths like gnome2_icon_savelist did, definitely not a full list of all icons. And the conditionals then adapted to work with that in postinst and postrm. This assumes I'm reading the code right - I just see it's similar code to the MIMEINFOFILES, and that ends up with a list of all files.

Reply:
Your comments didn't get updated in the web UI and didn't see any of them before all my previous comments were already posted. From only the original comment in context, I thought you meant that it's not necessary, because it's not used anymore, but is used in postrm - but you meant that they get exported anyways, unless the variable is declared with a `local` prefix. And to remove all of the line, not just the change to it in this patch. That said, the XDG_ECLASS_MIMEINFOFILES array seems to get exported fine, but if it's unnecessary to explicitly export and it's safer with arrays to not do so, then sure, lets remove it.